build.gradle.kts

```diff
@@ -763,44 +771,89 @@ project("SpongeVanilla") {
-        val depJoiner = StringJoiner(",\n", "  [\n", "\n  ]\n")
-
-        val foundConfig = project.configurations.findByName(configuration)
-        if (foundConfig == null) {
-            return
-        }
-        foundConfig.resolvedConfiguration.getFirstLevelModuleDependencies()
-                .filter { dependency -> dependency.getModuleName() != "SpongeAPI" }
-                .forEach { dependency ->
-                    //Get file input stream for reading the file content
-                    val depFile = dependency.getModuleArtifacts().filter { it.file.path.endsWith(".jar") }.first().file
-                    //Get file input stream for reading the file content
-                    val fis = FileInputStream(depFile)
-                    val md5Hash = DigestUtils.md5Hex(fis)
-                    fis.close()
-
-                    val depBuilder = StringJoiner(",\n", "    {\n", "\n    }")
-                    depBuilder.add("      \"group\": \"${dependency.getModuleGroup()}\"")
-                    depBuilder.add("      \"module\": \"${dependency.getModuleName()}\"")
-                    depBuilder.add("      \"version\": \"${dependency.getModuleVersion()}\"")
-                    depBuilder.add("      \"md5\": \"$md5Hash\"")
-                    depJoiner.add(depBuilder.toString())
+        val foundConfig = if (this.excludeConfiguration.isPresent) {
+            val config = project.configurations.detachedConfiguration(*this.configuration.get().allDependencies.toTypedArray())
+            val excludes = this.excludeConfiguration.get()
+            excludes.allDependencies.forEach {
+                config.exclude(group = it.group, module = it.name)
+            }
+            config
+        } else {
+            this.configuration.get()
+        }
+
+        val manifest = foundConfig.resolvedConfiguration.firstLevelModuleDependencies.asSequence()
+                .flatMap { it.allModuleArtifacts.asSequence() }
+                // only jars with the allowed classifiers
+                .filter { it.extension == "jar" && allowedClassifiers.get().contains(it.classifier ?: "") }
+                .filter { it.moduleVersion.id.name != "SpongeAPI" }
+                .distinct()
+                .map { dependency ->
+
+                    val ident = dependency.moduleVersion.id
+                    val version = (dependency.id.componentIdentifier as? ModuleComponentIdentifier)?.version ?: ident.version
+                    // Get file input stream for reading the file content
+                    val md5hash = dependency.file.inputStream().use {
+                        DigestUtils.md5Hex(it)
+                    }
+
+                    // create descriptor
+                    DependencyDescriptor(
+                            group = ident.group,
+                            module = ident.name,
+                            version = version, // TODO: can we get timestamped snapshot versions?
```

Comment:
don't need the timestamped snapshot versions, the sonatype search uses the maven.version specifically, not the timestamp'ed snapshot.